tests/test_report_writing.py

```diff
@@ -30,11 +47,89 @@ def test_school_name_replaced_in_paragraphs(data_path, templates_dir, temp_out_d
+    assert (example_school_data["entry_date"] < "2022-8-1").all()
+    assert (
+        (example_school_data["exit_date"].isna()) | (example_school_data["exit_date"] > "2021-7-31") | (example_school_data["exit_date"] < "2022-8-1")
+    ).all()
+
+    # test_1_2021-22, Discontinued and within date boundary, no month3 or month 6 results. Should show in table 1,2,3,4,5.
+    pupil_1 = test_filter_for_one_two_five.loc[test_filter_for_one_two_five.pupil_no == "test_1_2021-22"]
+    assert ((pupil_1["entry_date"] < "2022-8-1") | (pupil_1["entry_date"] > "2021-7-31")).all()
+    assert ((pupil_1["exit_date"] < "2022-8-1") | (pupil_1["exit_date"] > "2021-7-31") | (pupil_1["exit_date"].isna())).all()
+
+    # test_2_2021-22, Discontinued, not within date boundary, month3 testdate out of range, no month6. Should not be in any tables.
+    pupil_2 = test_filter_for_one_two_five.loc[test_filter_for_one_two_five.pupil_no == "test_2_2021-22"]
+    assert ((~pupil_2["entry_date"] < "2022-8-1") | (~pupil_2["entry_date"] > "2021-7-31")).all()
+    assert ((~pupil_2["exit_date"] < "2022-8-1") | (~pupil_2["exit_date"] > "2021-7-31") | (pupil_2["exit_date"].isna())).all()
+
+    # test_3_2021-22, Discontinued, entry date IN report date but exit date OUT of report date, no month3 or month 6 results. Should be in 1,2,5.
+    pupil_3 = test_filter_for_one_two_five.loc[test_filter_for_one_two_five.pupil_no == "test_3_2021-22"]
+    assert ((pupil_3["entry_date"] < "2022-8-1") | (pupil_3["entry_date"] > "2021-7-31")).all()
+    assert ((pupil_3["exit_date"] > "2022-8-1") | (pupil_3["exit_date"] < "2021-7-31") | (pupil_3["exit_date"].isna())).all()
+
+    # test_4_2021-22, Discontinued, entry date OUT of report date but exit date IN report date, month3 testdate in range. Should be in all tables.
+    pupil_4 = test_filter_for_one_two_five.loc[test_filter_for_one_two_five.pupil_no == "test_4_2021-22"]
+    assert ((pupil_4["entry_date"] > "2022-8-1") | (pupil_4["entry_date"] < "2021-7-31")).all
+    assert ((pupil_4["exit_date"] < "2022-8-1") | (pupil_4["exit_date"] > "2021-7-31") | (pupil_4["exit_date"].isna())).all()
+
+    # test_5_2021-22, Incomplete and within date boundary. Should be in 1,2,5.
+    pupil_5 = test_filter_for_one_two_five.loc[test_filter_for_one_two_five.pupil_no == "test_5_2021-22"]
+    assert ((pupil_5["entry_date"] > "2022-8-1") | (pupil_5["entry_date"] < "2021-7-31")).all
+    assert ((pupil_5["exit_date"] < "2022-8-1") | (pupil_5["exit_date"] > "2021-7-31") | (pupil_5["exit_date"].isna())).all()
+
+    # test_6_2021-22, Ongoing and entry date within boundary and NA exit date. Should be in 1,2,5.
+    pupil_6 = test_filter_for_one_two_five.loc[test_filter_for_one_two_five.pupil_no == "test_6_2021-22"]
+    assert ((pupil_6["entry_date"] < "2022-8-1") | (pupil_6["entry_date"] > "2021-7-31")).all()
+    assert (pupil_6["exit_date"].isna()).all()
```

Comment:
might make sense to make these separate tests (maybe with `test_filter_for_one_two_five` as a fixture?)